src/main.ts

```diff
@@ -380,51 +380,68 @@ function makeTimestampMethods(options: Options, longs: ReturnType<typeof makeLon
 
   const toTimestamp = conditionalOutput(
     'toTimestamp',
-    code`
-      function toTimestamp(date: Date): ${Timestamp} {
-        const seconds = ${seconds};
-        const nanos = (date.getTime() % 1_000) * 1_000_000;
-        return { seconds, nanos };
-      }
-    `
+    options.useDate === DateOption.STRING
+      ? code`
+          function toTimestamp(dateStr: string): ${Timestamp} {
+            const date = new Date(dateStr);
```

Comment:
I felt this was not smart because the variable named `date` can't be changed due to stuff outside of this function: https://github.com/eqyiel/ts-proto/blob/c4977a2a19ad565766d581c4e3d733d1667de150/src/main.ts#L371-L379

I don't think it's _that_ much worse though, because that was already the case 😇